python_modules/dagster/dagster/_core/definitions/automation_condition_sensor_definition.py

```diff
@@ -57,6 +58,11 @@ def _evaluate(sensor_def: "AutomationConditionSensorDefinition", context: Sensor
             "or split this sensor into multiple AutomationConditionSensorDefinitions with AssetSelections that target fewer "
             "assets or checks."
         )
+    if not context.instance.auto_materialize_use_sensors:
+        raise DagsterInvalidInvocationError(
+            "Cannot evaluate an AutomationConditionSensorDefinition if the instance setting "
+            " `auto_materialize: use_sensors` is set to False. Update your configuration to prevent this error.",
+        )
```

Comment:
I don't think this is going to work the way we intend here when executed in cloud user code. The source of truth for this in cloud is a deployment setting, which works fine when it is being checked in a daemon, but is not currently propagated to cloud user code. Right now the current implementation would look for the same dagster.yaml key that you would set in dagster OSS.

Reply:
good callout, I've moved this logic to be inside the sensor daemon and added a test for the behavior